pkgs/tools/networking/zerotierone/0001-darwin-disable-link-time-optimization.patch

```diff
@@ -0,0 +1,32 @@
+From f71d88a15f537cbc5ea80ad2922c013553a5caab Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?J=C3=B6rg=20Thalheim?= <joerg@thalheim.io>
+Date: Fri, 21 Jun 2024 14:43:03 +0200
+Subject: [PATCH 2/2] darwin: disable link-time optimization
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+It's not yet clear to me, but in our build we somehow mix objects with
+and without flto.
+
+Signed-off-by: Jörg Thalheim <joerg@thalheim.io>
+---
+ make-mac.mk | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/make-mac.mk b/make-mac.mk
+index 7af200ad..b388c05a 100644
+--- a/make-mac.mk
++++ b/make-mac.mk
+@@ -84,7 +84,7 @@ ifeq ($(ZT_DEBUG),1)
+ node/Salsa20.o node/SHA512.o node/C25519.o node/Poly1305.o: CFLAGS = -Wall -O2 -g $(INCLUDES) $(DEFS)
+ else
+ 	CFLAGS?=-Ofast -fstack-protector-strong
+-	CFLAGS+=$(ARCH_FLAGS) -Wall -flto -fPIE -mmacosx-version-min=$(MACOS_VERSION_MIN) -DNDEBUG -Wno-unused-private-field $(INCLUDES) $(DEFS)
++	CFLAGS+=$(ARCH_FLAGS) -Wall -fPIE -mmacosx-version-min=$(MACOS_VERSION_MIN) -DNDEBUG -Wno-unused-private-field $(INCLUDES) $(DEFS)
```

Comment:
I believe this is needed because `cargo build` doesn't do link time optimization, but I am unsure how to enable it.

Reply:
I am open to other suggestions, but it wouldn't be the end of the world either.

Reply:
The effect is that most object files are in bitcode format but some are not and the linker doesn't like that.

Reply:
`cargo build` does do `lto` if requested in the `Cargo.toml` [see the cargo book][1]. Tl;dr: set `profile.release.lto = true or "thin"` (where `thin` is faster but may achieve less performance gains).

[1]: https://doc.rust-lang.org/cargo/reference/profiles.html#lto

Reply:
Can I do this from cli?

Reply:
AFAIK no through cargo, but maybe `RUSTFLAGS='-C lto=thin` is enough (the lto setting in Cargo.toml does more that just set `RUSTFLAGS`) https://doc.rust-lang.org/cargo/reference/profiles.html#lto